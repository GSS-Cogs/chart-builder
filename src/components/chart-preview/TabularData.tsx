// @ts-ignore
import React, { useCallback, useEffect, useState } from "react";
// import Plotly from "plotly.js-basic-dist-min";
// import createPlotlyComponent from "react-plotly.js/factory";
import Table from "../table/Table";

const TabularData = ({ chartDefinition, selectedColumns }: any) => {
  let { data, layout, config, chartType } = chartDefinition;

  useEffect(() => {}, []);

  return (
    <Table selectedColumns={selectedColumns} data={data} rowsPerPage={10} />
  );
};

export default TabularData;
