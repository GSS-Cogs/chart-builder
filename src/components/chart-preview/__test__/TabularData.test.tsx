import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
// import configureStore from "redux-mock-store";
// import { Provider } from "react-intl-redux";

import TabularData from "../TabularData";
import { multipleSeriesChartDefinition } from "./data/multipleSeriesChartDefinition";
import { mapChartDefinition } from "./data/mapChartDefinition";

describe("TabularData", () => {
  it("renders the TabularData component", () => {
    const data = multipleSeriesChartDefinition;
    const selectedColumns = ["week_starting"];
    const component = render(
      <TabularData chartDefinition={data} selectedColumns={selectedColumns} />,
    );
    const json = component.container;
    expect(json).toMatchSnapshot();
  });

  it("renders the TabularData component and scrolls through pagination pages", () => {
    const data = multipleSeriesChartDefinition;
    const selectedColumns = ["week_starting"];
    const component = render(
      <TabularData chartDefinition={data} selectedColumns={selectedColumns} />,
    );

    for (let i = 2; i <= 5; i++) {
      const button = screen.getByText(i.toString());
      fireEvent.click(button);

      const json = component.container;
      expect(json).toMatchSnapshot();
    }
  });

  it("renders the TabularData component with given map chartDefintion data", () => {
    const data = mapChartDefinition;
    const selectedColumns = [""];
    const component = render(
      <TabularData chartDefinition={data} selectedColumns={selectedColumns} />,
    );
    let json = component.container;
    expect(json).toMatchSnapshot();

    const button = screen.getByText("38");
    fireEvent.click(button);

    json = component.container;
    expect(json).toMatchSnapshot();
  });
});
