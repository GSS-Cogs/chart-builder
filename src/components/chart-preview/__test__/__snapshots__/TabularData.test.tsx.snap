// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabularData renders the TabularData component 1`] = `
<div>
  <div
    class="tableWrapper"
  >
    <table
      class="table"
    >
      <thead
        class="tableRowHeader"
      >
        <tr>
          <th
            class="tableHeader"
          >
            week_starting
          </th>
          <th
            class="tableHeader"
          >
            England
          </th>
          <th
            class="tableHeader"
          >
            Scotland
          </th>
          <th
            class="tableHeader"
          >
            Wales
          </th>
          <th
            class="tableHeader"
          >
            Northern Ireland
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            0.1
          </td>
          <td
            class="tableCell"
          >
            13/09/2021
          </td>
          <td
            class="tableCell"
          >
            13/09/2021
          </td>
          <td
            class="tableCell"
          >
            13/09/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            0.3
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            13/09/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            0.4
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            20/09/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            0.6
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            20/09/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            0.7
          </td>
          <td
            class="tableCell"
          >
            20/09/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            27/09/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            1.1
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            20/09/2021
          </td>
          <td
            class="tableCell"
          >
            04/10/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            2.2
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            11/10/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            2.5
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            27/09/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            3
          </td>
          <td
            class="tableCell"
          >
            27/09/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            4.3
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            27/09/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="tableFooter"
  >
    <div>
      <button
        class="endButton inactiveButton"
        disabled=""
      >
        &lt; Prev
      </button>
      <button
        class="button buttonNormalWidth activeButton"
      >
        1
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        2
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        3
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        4
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        5
      </button>
      <button
        class="endButton inactiveButton"
      >
        Next &gt;
      </button>
    </div>
    <div
      class="results"
    >
      Showing results 
      1 - 10
       
       of 50 results
    </div>
  </div>
</div>
`;

exports[`TabularData renders the TabularData component and scrolls through pagination pages 1`] = `
<div>
  <div
    class="tableWrapper"
  >
    <table
      class="table"
    >
      <thead
        class="tableRowHeader"
      >
        <tr>
          <th
            class="tableHeader"
          >
            week_starting
          </th>
          <th
            class="tableHeader"
          >
            England
          </th>
          <th
            class="tableHeader"
          >
            Scotland
          </th>
          <th
            class="tableHeader"
          >
            Wales
          </th>
          <th
            class="tableHeader"
          >
            Northern Ireland
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            4.5
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            18/10/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            5.6
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            04/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            6.1
          </td>
          <td
            class="tableCell"
          >
            04/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            8.1
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            04/10/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            9.2
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            11/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            25/10/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            9.7
          </td>
          <td
            class="tableCell"
          >
            11/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            12.2
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            11/10/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            13.3
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            18/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            13.6
          </td>
          <td
            class="tableCell"
          >
            18/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            16.1
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            18/10/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="tableFooter"
  >
    <div>
      <button
        class="endButton inactiveButton"
      >
        &lt; Prev
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        1
      </button>
      <button
        class="button buttonNormalWidth activeButton"
      >
        2
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        3
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        4
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        5
      </button>
      <button
        class="endButton inactiveButton"
      >
        Next &gt;
      </button>
    </div>
    <div
      class="results"
    >
      Showing results 
      11 - 20
       
       of 50 results
    </div>
  </div>
</div>
`;

exports[`TabularData renders the TabularData component and scrolls through pagination pages 2`] = `
<div>
  <div
    class="tableWrapper"
  >
    <table
      class="table"
    >
      <thead
        class="tableRowHeader"
      >
        <tr>
          <th
            class="tableHeader"
          >
            week_starting
          </th>
          <th
            class="tableHeader"
          >
            England
          </th>
          <th
            class="tableHeader"
          >
            Scotland
          </th>
          <th
            class="tableHeader"
          >
            Wales
          </th>
          <th
            class="tableHeader"
          >
            Northern Ireland
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            17
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            01/11/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            17.4
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            25/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            17.8
          </td>
          <td
            class="tableCell"
          >
            25/10/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            19.1
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            25/10/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            22.4
          </td>
          <td
            class="tableCell"
          >
            01/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            22.5
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            01/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            23
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            01/11/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            26
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            08/11/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            26.8
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            08/11/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            27.4
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            08/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="tableFooter"
  >
    <div>
      <button
        class="endButton inactiveButton"
      >
        &lt; Prev
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        1
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        2
      </button>
      <button
        class="button buttonNormalWidth activeButton"
      >
        3
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        4
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        5
      </button>
      <button
        class="endButton inactiveButton"
      >
        Next &gt;
      </button>
    </div>
    <div
      class="results"
    >
      Showing results 
      21 - 30
       
       of 50 results
    </div>
  </div>
</div>
`;

exports[`TabularData renders the TabularData component and scrolls through pagination pages 3`] = `
<div>
  <div
    class="tableWrapper"
  >
    <table
      class="table"
    >
      <thead
        class="tableRowHeader"
      >
        <tr>
          <th
            class="tableHeader"
          >
            week_starting
          </th>
          <th
            class="tableHeader"
          >
            England
          </th>
          <th
            class="tableHeader"
          >
            Scotland
          </th>
          <th
            class="tableHeader"
          >
            Wales
          </th>
          <th
            class="tableHeader"
          >
            Northern Ireland
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            27.8
          </td>
          <td
            class="tableCell"
          >
            08/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            30.2
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            15/11/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            31.5
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            15/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            33.4
          </td>
          <td
            class="tableCell"
          >
            15/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            22/11/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            34.8
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            15/11/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            36.5
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            22/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            38
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            29/11/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            38.8
          </td>
          <td
            class="tableCell"
          >
            22/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            41.6
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            29/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            42.4
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            22/11/2021
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="tableFooter"
  >
    <div>
      <button
        class="endButton inactiveButton"
      >
        &lt; Prev
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        1
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        2
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        3
      </button>
      <button
        class="button buttonNormalWidth activeButton"
      >
        4
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        5
      </button>
      <button
        class="endButton inactiveButton"
      >
        Next &gt;
      </button>
    </div>
    <div
      class="results"
    >
      Showing results 
      31 - 40
       
       of 50 results
    </div>
  </div>
</div>
`;

exports[`TabularData renders the TabularData component and scrolls through pagination pages 4`] = `
<div>
  <div
    class="tableWrapper"
  >
    <table
      class="table"
    >
      <thead
        class="tableRowHeader"
      >
        <tr>
          <th
            class="tableHeader"
          >
            week_starting
          </th>
          <th
            class="tableHeader"
          >
            England
          </th>
          <th
            class="tableHeader"
          >
            Scotland
          </th>
          <th
            class="tableHeader"
          >
            Wales
          </th>
          <th
            class="tableHeader"
          >
            Northern Ireland
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            43.9
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            06/12/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            45.2
          </td>
          <td
            class="tableCell"
          >
            29/11/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            45.9
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            06/12/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            47.6
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            29/11/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            50.1
          </td>
          <td
            class="tableCell"
          >
            06/12/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            51.8
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            13/12/2021
          </td>
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            52.4
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            06/12/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            53.6
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            13/12/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            58.8
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          >
            13/12/2021
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            61.3
          </td>
          <td
            class="tableCell"
          >
            13/12/2021
          </td>
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
          <td
            class="tableCell"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="tableFooter"
  >
    <div>
      <button
        class="endButton inactiveButton"
      >
        &lt; Prev
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        1
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        2
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        3
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        4
      </button>
      <button
        class="button buttonNormalWidth activeButton"
      >
        5
      </button>
      <button
        class="endButton inactiveButton"
        disabled=""
      >
        Next &gt;
      </button>
    </div>
    <div
      class="results"
    >
      Showing results 
      41 - 50
       
       of 50 results
    </div>
  </div>
</div>
`;

exports[`TabularData renders the TabularData component with given map chartDefintion data 1`] = `
<div>
  <div
    class="tableWrapper"
  >
    <table
      class="table"
    >
      <thead
        class="tableRowHeader"
      >
        <tr>
          <th
            class="tableHeader"
          >
            Location
          </th>
          <th
            class="tableHeader"
          >
            Va
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Aberdeen City
          </td>
          <td
            class="tableCell"
          >
            1155
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Aberdeenshire
          </td>
          <td
            class="tableCell"
          >
            1707
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Adur
          </td>
          <td
            class="tableCell"
          >
            209
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Allerdale
          </td>
          <td
            class="tableCell"
          >
            637
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Amber Valley
          </td>
          <td
            class="tableCell"
          >
            663
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Angus
          </td>
          <td
            class="tableCell"
          >
            679
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Antrim and Newtownabbey
          </td>
          <td
            class="tableCell"
          >
            947
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Ards and North Down
          </td>
          <td
            class="tableCell"
          >
            662
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Argyll and Bute
          </td>
          <td
            class="tableCell"
          >
            267
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Armagh City, Banbridge and Craigavon
          </td>
          <td
            class="tableCell"
          >
            1583
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="tableFooter"
  >
    <div>
      <button
        class="endButton inactiveButton"
        disabled=""
      >
        &lt; Prev
      </button>
      <button
        class="button buttonNormalWidth activeButton"
      >
        1
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        2
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        3
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        4
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        5
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        6
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        7
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        ...
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        38
      </button>
      <button
        class="endButton inactiveButton"
      >
        Next &gt;
      </button>
    </div>
    <div
      class="results"
    >
      Showing results 
      1 - 10
       
       of 379 results
    </div>
  </div>
</div>
`;

exports[`TabularData renders the TabularData component with given map chartDefintion data 2`] = `
<div>
  <div
    class="tableWrapper"
  >
    <table
      class="table"
    >
      <thead
        class="tableRowHeader"
      >
        <tr>
          <th
            class="tableHeader"
          >
            Location
          </th>
          <th
            class="tableHeader"
          >
            Va
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Wokingham
          </td>
          <td
            class="tableCell"
          >
            726
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Wolverhampton
          </td>
          <td
            class="tableCell"
          >
            925
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Worcester
          </td>
          <td
            class="tableCell"
          >
            333
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Worthing
          </td>
          <td
            class="tableCell"
          >
            315
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Wrexham
          </td>
          <td
            class="tableCell"
          >
            894
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Wychavon
          </td>
          <td
            class="tableCell"
          >
            958
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Wyre
          </td>
          <td
            class="tableCell"
          >
            686
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            Wyre Forest
          </td>
          <td
            class="tableCell"
          >
            386
          </td>
        </tr>
        <tr
          class="tableRowItems"
        >
          <td
            class="tableCell"
          >
            York
          </td>
          <td
            class="tableCell"
          >
            808
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="tableFooter"
  >
    <div>
      <button
        class="endButton inactiveButton"
      >
        &lt; Prev
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        1
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        ...
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        32
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        33
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        34
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        35
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        36
      </button>
      <button
        class="button buttonNormalWidth inactiveButton"
      >
        37
      </button>
      <button
        class="button buttonNormalWidth activeButton"
      >
        38
      </button>
      <button
        class="endButton inactiveButton"
        disabled=""
      >
        Next &gt;
      </button>
    </div>
    <div
      class="results"
    >
      Showing results 
      371 - 379
       
       of 379 results
    </div>
  </div>
</div>
`;
